src/lib.rs

```diff
@@ -208,6 +208,21 @@ impl<S: Clone> Context<S> {
             state: state,
         }
     }
+
+    /// Parses route-specified params to a value
+    pub fn route_params<P>(&self) -> StdResult<P, serde_urlencoded::de::Error>
+        where for<'a> P: serde::Deserialize<'a> {
+        self.route.params()
+    }
+
+    /// Parses query params to a value
+    pub fn query_params<P>(&self) -> StdResult<P, Box<Error>>
+        where for<'a> P: serde::Deserialize<'a> {
+        //TODO figure out how to use non-Box<Error> error here
+        let query_params_string = self.req.query().ok_or("no query params")?;
```

Comment:
What if there is no query string but every field on P is an Option? Maybe if there is no query we just use empty string and let serde fail if it wants to.

Reply:
Good call!